src/commands/org/logout.ts

```diff
@@ -0,0 +1,236 @@
+  public static readonly description = messages.getMessage('description');
+  public static readonly examples = messages.getMessages('examples');
+
+  public static readonly deprecateAliases = true;
+  public static aliases = ['force:auth:logout', 'auth:logout'];
+
+  public static readonly flags = {
+    'target-org': Flags.string({
+      summary: messages.getMessage('flags.target-org.summary'),
+      char: 'o',
+      aliases: ['targetusername', 'u'],
+      deprecateAliases: true,
+    }),
+    all: Flags.boolean({
+      char: 'a',
+      summary: messages.getMessage('flags.all.summary'),
+      description: messages.getMessage('flags.all.description'),
+      required: false,
+      default: false,
+      exclusive: ['target-org'],
+    }),
+    'no-prompt': Flags.boolean({
+      char: 'p',
+      summary: commonMessages.getMessage('flags.no-prompt.summary'),
+      required: false,
+      deprecateAliases: true,
+      aliases: ['noprompt'],
+    }),
+    loglevel,
+  };
+
+  private flags: Interfaces.InferredFlags<typeof Logout.flags>;
+
+  private static buildChoices(orgAuths: OrgAuthorization[], all: boolean): Array<choice | Separator> {
+    const maxUsernameLength = Math.max('Username'.length, ...orgAuths.map((orgAuth) => orgAuth.username.length));
+    const maxAliasLength = Math.max(
+      'Aliases'.length,
+      ...orgAuths.map((orgAuth) => (orgAuth.aliases ? orgAuth.aliases.join(',') : '').length)
+    );
+    const maxConfigLength = Math.max(
+      'Configs'.length,
+      ...orgAuths.map((orgAuth) => (orgAuth.configs ? orgAuth.configs.join(',') : '').length)
+    );
+    const maxTypeLength = Math.max(
+      'Type'.length,
+      ...orgAuths.map((orgAuth) => {
+        if (orgAuth.isScratchOrg) {
+          return 'Scratch'.length;
+        }
+        if (orgAuth.isDevHub) {
+          return 'DevHub'.length;
+        }
+        if (orgAuth.isSandbox) {
+          return 'Sandbox'.length;
+        }
+        return 0;
+      })
+    );
+    const choices = orgAuths
+      .map((orgAuth) => {
+        const aliasString = (orgAuth.aliases ? orgAuth.aliases.join(',') : '').padEnd(maxAliasLength, ' ');
+        const configString = (orgAuth.configs ? orgAuth.configs.join(',') : '').padEnd(maxConfigLength, ' ');
+        const typeString = chalk.dim(
+          (orgAuth.isScratchOrg ? 'Scratch' : orgAuth.isDevHub ? 'DevHub' : orgAuth.isSandbox ? 'Sandbox' : '').padEnd(
+            maxTypeLength,
+            ' '
+          )
+        );
+        // username - aliases - configs
+        const key = `${chalk.bold(
+          orgAuth.username.padEnd(maxUsernameLength)
+        )} | ${typeString} | ${aliasString} | ${chalk.yellowBright(configString)}`;
+        return { name: key, value: orgAuth, checked: all, short: `${os.EOL}${orgAuth.username}` };
+      })
+      .sort((a, b) => a.value.username.localeCompare(b.value.username));
+    const userHeader = `${'Username'.padEnd(maxUsernameLength, ' ')}`;
```

Comment:
too bad we have to do all of this in the plugin

Reply:
➕